### ISSUE TITLE
Check failure when running array_ops.quantize_and_dequantize_v2 probably due to dimension mismatch.

### ISSUE THREAD

**Issue by nimashiri** [tensorflow/tensorflow#59169]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
Check failure when running .quantize_and_dequantize_v2 probably due to dimension mismatch.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import array_ops
try:
  arg_0_tensor = tf.saturate_cast(tf.random.uniform([2, 3, 4], minval=0, maxval=2, dtype=tf.int64), dtype=tf.uint64)
  arg_0 = tf.identity(arg_0_tensor)
  arg_1 = -1
  arg_2 = -1
  range_given = True
  round_mode = "nan"
  axis = None
  out = array_ops.quantize_and_dequantize_v2(arg_0,arg_1,arg_2,range_given=range_given,round_mode=round_mode,axis=axis,)
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
Running on Crash oracle!
2023-01-08 19:24:06.219135: F ./tensorflow/python/eager/pywrap_tensor_conversion.h:58] Check failed: !PyErr_Occurred() 
Aborted

```
```
</details>

**Comment by mohantym** [comment]
@nimashiri !
Thanks for reporting the bug on quantize_and_dequantize_v2.

@SuryanarayanaY !
Could you look at this issue . Attached gist in [2.10](https://colab.sandbox.google.com/gist/mohantym/7322d2ba62ec0f699cc68df024237050/git_59169.ipynb#scrollTo=JerqUqYz0Z6_), [2.11](https://colab.sandbox.google.com/gist/mohantym/280ce18a0d5c48d12b20f3c5ed50374d/git_59169.ipynb#scrollTo=HVzfe1R31gKQ) and [nightly](https://colab.sandbox.google.com/gist/mohantym/63d5c15b59897ccebeefdcdb4bf2b975/git_59169.ipynb#scrollTo=JerqUqYz0Z6_) for reference.

Thank you!

**Comment by SuryanarayanaY** [comment]
Hi @nimashiri, 
I have checked with similar high level API `tf.quantization.quantize_and_dequantize_v2` and observed similar error(check fail) as reported.

But the check fails are expected as all the Ops not validated to all the Scenarios as a tradeoff for Performance.For example for this high level API the arguments are clearly mentioned and if we follow the documentation we can remove the error with valid arguments.Please refer to attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/1669fc79acbdf42cd3d79765ce013dab/59169-2-11v_r1.ipynb).

Please refer to the below [commit](https://github.com/tensorflow/tensorflow/commit/620ab3e167b0a64b6732dac16fd0edaf8284cb8e) regarding Policy w.r.t Check fails and confirm if it falls under Vulnerability. AFAIK this only lead to program termination and no other
exploit.

**Comment by SuryanarayanaY** [comment]
The issue got resolved in tf-nightly (2.12.0-dev20230130).Executed the code multiple times and got expected error.Please refer to attached snapshot below.Hence the issue not reopening.
<img width="1503" alt="Screenshot 2023-01-30 at 10 05 44 PM" src="https://user-images.githubusercontent.com/116063290/215539424-6d658630-7292-4857-a347-5327345792f9.png">
 
@nimashiri FYIP. If anybody still observed check fail please confirm and the issue can be reopened.
